runtime/interpreter/interpreter.go

```diff
@@ -4483,6 +4483,15 @@ func (interpreter *Interpreter) UseMemory(usage common.MemoryUsage) {
 	interpreter.memoryGauge.UseMemory(usage)
 }
 
+// UseKnownMemory uses a pre-determined amount of memory
+//
+func (interpreter *Interpreter) UseKnownMemory(kind common.MemoryKind, amount uint64) {
+	interpreter.UseMemory(common.MemoryUsage{
+		Kind:   kind,
+		Amount: amount,
+	})
+}
```

Comment:
All of the simple types use `1` for `Amount` too. Since this is such a common use case, maybe a more specific version of this function would be better.

```suggestion
// UseConstantMemory uses a pre-determined amount of memory
//
func (interpreter *Interpreter) UseConstantMemory(kind common.MemoryKind) {
	interpreter.UseMemory(common.MemoryUsage{
		Kind:   kind,
		Amount: uint64(1),
	})
}
```